app/controllers/groups_controller.php

```diff
@@ -874,8 +908,36 @@ function _addGroupByImport($lines, $courseId, $update, $identifier, $render=true
      * @access public
      * @return void
      */
-    function export($courseId)
+    function export($courseId, $exportTo = 'file')
     {
+        $this->set('exportTo', $exportTo);
+
+        if ($exportTo == 'canvas') {
+
+            App::import('Component', 'CanvasCourse');
+
+            $this->data['Group']['syncType'] = 'export';
+            $this->data['Group']['updateCanvasGroups'] = true;
+            $this->data['Group']['selectAllGroups'] = true;
+
+            $this->syncCanvas($courseId);
```

Comment:
During export, if the student is not enrolled in Canvas course, he/she will be silently ignored and the screen will display export successful.  Although on the export page, the instructions stated that `Any students in iPeer that are not in Canvas will not be exported.`, should we include a warning on the export result page that some group members are not exported to Canvas?

Reply:
Tried to export groups to Canvas with students not in the Canvas course (but have valid Canvas a/c), and the sync will generate the following flash messages.  Should the students be auto-enrolled to the Canvas course?  If not, should the error message says the students are not enrolled to the Canvas course instead of not having associated Canvas a/c?

```
Only some of the requested groups and users were exported to Canvas. Those are highlighted below.

The following errors happened:

redshirt0002 was not added to the group in Canvas because they do not have an associated Canvas account.
If you continue having issues, please contact support.
```